cmd/provisioner-localpv/app/config.go

```
@@ -0,0 +1,217 @@
+	// a hostpath ( sub directory or a storage device)
+	KeyPVStorageType = "StorageType"
+	//KeyPVBasePath defines base directory for hostpath volumes
+	// can be configured via the StorageClass annotations.
+	KeyPVBasePath = "BasePath"
+	//KeyPVRelativePath defines the alternate folder name under the BasePath
+	// By default, the pv name will be used as the folder name.
+	// KeyPVBasePath can be useful for providing the same underlying folder
+	// name for all replicas in a Statefulset.
+	// Will be a property of the PVC annotations.
+	KeyPVRelativePath = "RelativePath"
+	//KeyPVAbsolutePath specifies a complete hostpath instead of
+	// auto-generating using BasePath and RelativePath. This option
+	// is specified with PVC and is useful for granting shared access
+	// to underlying hostpaths across multiple pods.
+	KeyPVAbsolutePath = "AbsolutePath"
+)
+
+const (
+	// Some of the PVCs launched with older helm charts, still
+	// refer to the StorageClass via beta annotations.
+	betaStorageClassAnnotation = "volume.beta.kubernetes.io/storage-class"
+)
+
+//getVolumeConfig creates a new VolumeConfig struct by
+// parsing and merging the configuration provided in the PVC
+// annotation - cas.openebs.io/config with the
+// default configuration of the provisioner.
+func (p *Provisioner) GetVolumeConfig(pvName string, pvc *v1.PersistentVolumeClaim) (*VolumeConfig, error) {
+
+	pvConfig := p.defaultConfig
+
+	//Fetch the SC
+	scName := GetStorageClassName(pvc)
+	sc, err := p.kubeClient.StorageV1().StorageClasses().Get(*scName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get storageclass: missing sc name {%v}", scName)
+	}
+
+	// extract and merge the cas config from storageclass
+	scCASConfigStr := sc.ObjectMeta.Annotations[string(mconfig.CASConfigKey)]
+	glog.Infof("SC %v has config:%v", *scName, scCASConfigStr)
+	if len(strings.TrimSpace(scCASConfigStr)) != 0 {
+		scCASConfig, err := cast.UnMarshallToConfig(scCASConfigStr)
```

This `err` is swallowed.

---

good one. fixed.